src/gui/guiOptionsChange.cpp

```diff
@@ -0,0 +1,235 @@
+	const wchar_t *text = wgettext(name.c_str());
+	Environment->addCheckBox(g_settings->getBool(setting), rect, this, ID, text);
+	delete[] text;
+}
+
+void GUIOptionsChange::addSlider(int ID, int max, int min, int init, int xoff, int yoff)
+{
+	core::rect<s32> rect(0, 0, 300, 20);
+	rect = rect + v2s32(m_size.X / 2 + xoff, m_size.Y / 2 + yoff);
+	gui::IGUIScrollBar *e = Environment->addScrollBar(true, rect, this, ID);
+	e->setMax(max);
+	e->setMin(min);
+	e->setPos(init);
+}
+
+void GUIOptionsChange::addDynText(const std::string& name, int value, const std::string& ending, int ID, int xoff, int yoff)
+{
+	core::rect<s32> rect(0, 0, 160, 20);
+	rect = rect + v2s32(m_size.X / 2 + xoff, m_size.Y / 2 + yoff);
+
+	const wchar_t *text = wgettext(name.c_str());
+	core::stringw temp_text = text;
+	delete[] text;
+
+	temp_text += core::stringw(value) + core::stringw(wgettext(ending.c_str()));
+	Environment->addStaticText(temp_text.c_str(), rect, false,
+			true, this, ID);
+}
+
+void GUIOptionsChange::addText(const std::string& name, int ID, int xoff, int yoff)
+{
+	core::rect<s32> rect(0, 0, 160, 20);
+	rect = rect + v2s32(m_size.X / 2 + xoff, m_size.Y / 2 + yoff);
+
+	const wchar_t *text = wgettext(name.c_str());
+	core::stringw temp_text = text;
+	delete[] text;
+	
+	Environment->addStaticText(temp_text.c_str(), rect, false,
+			true, this, ID);
+}
+
+void GUIOptionsChange::updateDynText(const SEvent& event, const std::string& setting, int scale, int ID, const std::string& beginning, const std::string& ending)
+{
+	s32 pos = ((gui::IGUIScrollBar*)event.GUIEvent.Caller)->getPos();
+	g_settings->setFloat(setting, (float) pos / scale);
+
+	gui::IGUIElement *e = getElementFromId(ID);
+	const wchar_t *text = wgettext(beginning.c_str());
+	core::stringw temp_text = text;
+	delete[] text;
+
+	temp_text += core::stringw(pos) + core::stringw(ending.c_str());
+	e->setText(temp_text.c_str());
+}
+
+void GUIOptionsChange::regenerateGui(v2u32 screensize)
+{
+	/*
+		Remove stuff
+	*/
+	removeChildren();
+
+	/*
+		Calculate new sizes and positions
+	*/
+	float height = 230;
+	float width = 380;
+	int spacing = 30;
+	int column_pos = (int)(-width/2+spacing);
+	int row = (int)(-height/2+spacing);
+	DesiredRect = core::rect<s32>(
+		screensize.X/2 - width/2,
+		screensize.Y/2 - height/2,
+		screensize.X/2 + width/2,
+		screensize.Y/2 + height/2
+	);
+	recalculateAbsolutePosition(false);
+	m_size = DesiredRect.getSize();
+	int volume = (int)(g_settings->getFloat("sound_volume") * 100);
+	int FOV = (int)(g_settings->getFloat("fov"));
+	
+
+	//sound stuff
+	addDynText("Sound Volume: ", volume, "%", ID_soundText, column_pos, row);
+	addCheckBox("Mute", "mute_sound", ID_soundMuteButton, column_pos+230, row);
+	row+=spacing;
+	addSlider(ID_soundSlider, 100, 0, volume, column_pos, row);
+	//fov stuff
+	row+=spacing;
+	addDynText("Field Of View: ", FOV, "", ID_fovText, column_pos, row);
+	row+=spacing;
+	addSlider(ID_fovSlider, 160, 51, FOV, column_pos, row);
```

Comment:
Minimum can be 45 degrees, i tested the lowest that does not trigger loading of distant world.